platform-sdk/swirlds-merkledb/src/main/java/com/swirlds/merkledb/collections/AbstractLongList.java

```diff
@@ -221,15 +221,31 @@ protected AbstractLongList(
                 }
 
                 maxLongs = headerBuffer.getLong();
+
+                // Compute how many longs are in the file body
+                final long longsInFile = (fileChannel.size() - currentFileHeaderSize) / Long.BYTES;
+
                 if (formatVersion >= MIN_VALID_INDEX_SUPPORT_VERSION) {
-                    minValidIndex.set(headerBuffer.getLong());
-                    // "inflating" the size by number of indices that are to the left of the min valid index
-                    size.set(minValidIndex.get() + (fileChannel.size() - currentFileHeaderSize) / Long.BYTES);
+                    final long readMinValidIndex = headerBuffer.getLong();
+
+                    // If the file is empty or readMinValidIndex < 0, treat it as an empty list
+                    if (longsInFile <= 0 || readMinValidIndex < 0) {
+                        size.set(0);
+                        minValidIndex.set(-1);
+                        maxValidIndex.set(-1);
+                    } else {
+                        // Otherwise, compute the size by "inflating" it to include the number of indices to the left of
+                        // the min valid index.
+                        minValidIndex.set(readMinValidIndex);
+                        size.set(readMinValidIndex + longsInFile);
+                        maxValidIndex.set(size.get() - 1);
```

Comment:
I wonder if we need to store both `size` and `maxValidIndex`, if they are always off by 1?

Reply:
Not always. Size can be different if we update valid range and then add element(s) -- size will reflect real right boundary:
```
longList.updateValidRange(0, 5000);
longList.put(1, 1);
System.out.println(longList.size()); --> "2"
```

Reply:
cc @imalygin